hazelcast-command-line/src/main/java/com/hazelcast/commandline/HazelcastCommandLine.java

```diff
@@ -0,0 +1,195 @@
+            @Option(names = {"-d", "--daemon"},
+                    description = "Starts in daemon mode.")
+                    boolean daemon,
+            @Option(names = {"-v", "--verbose"},
+                    description = "Output with FINE level verbose logging.")
+                    boolean verbose,
+            @Option(names = {"-vv", "--vverbose"},
+                    description = "Output with FINEST level verbose logging.")
+                    boolean finestVerbose)
+            throws IOException, InterruptedException {
+        List<String> args = new ArrayList<>();
+        if (!isNullOrEmpty(configFilePath)) {
+            args.add("-Dhazelcast.config=" + configFilePath);
+        }
+        if (!isNullOrEmpty(port)) {
+            args.add("-Dnetwork.port=" + port);
+        }
+        if (!isNullOrEmpty(hzInterface)) {
+            args.add("-Dnetwork.interface=" + hzInterface);
+        }
+
+        args.add("-Djava.net.preferIPv4Stack=true");
+        addLogging(args, verbose, finestVerbose);
+
+        if (javaOptions != null && !javaOptions.isEmpty()) {
+            addJavaOptionsToArgs(javaOptions, args);
+        }
+
+        buildAndStartJavaProcess(args, additionalClassPath, daemon);
+    }
+
+    private void addJavaOptionsToArgs(List<String> javaOptions, List<String> args) {
+        Set<String> userProvidedArgs = args.stream()
+                .map(arg -> arg.split("=")[0])
+                .collect(Collectors.toSet());
+
+        for (String opt : javaOptions) {
+            if (!userProvidedArgs.contains(opt.split("=")[0])) {
+                args.add(opt);
+            }
+        }
+    }
+
+    private void buildAndStartJavaProcess(List<String> parameters, String[] additionalClassPath, boolean daemon)
+            throws IOException, InterruptedException {
+        List<String> commandList = new ArrayList<>();
+        StringBuilder classpath = new StringBuilder(System.getProperty("java.class.path"));
+        if (additionalClassPath != null) {
+            for (String path : additionalClassPath) {
+                classpath.append(CLASSPATH_SEPARATOR).append(path);
+            }
+        }
+        String path = System.getProperty("java.home") + "/bin/java";
+        commandList.add(path);
+        commandList.add("-cp");
+        commandList.add(classpath.toString());
+        commandList.addAll(parameters);
+        fixModularJavaOptions(commandList);
+        commandList.add(HazelcastMember.class.getName());
+        processExecutor.buildAndStart(commandList, getRedirect(daemon), daemon);
+    }
+
+    private ProcessBuilder.Redirect getRedirect(boolean daemon) {
+        Clock clock = Clock.systemDefaultZone();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd.HH.mm.ss").withZone(clock.getZone());
+        String formattedDate = formatter.format(clock.instant());
+        String logFile = WORKING_DIRECTORY + "/logs/hazelcast." + formattedDate + ".out";
```

Comment:
I think this causes more harm than good. Let's document that for the daemon mode there is no output and logging should be configured to log to a file (which it is by default).

Reply:
I was focused on keeping the script backward compatible. In this case, I agree that it's awkward.

Reply:
Removed stdout redirection to file, only error stream will be logged to the file:  https://github.com/hazelcast/hazelcast/pull/20262/commits/67ce4af4a3843932ae28263d1e36febdf269dcec